### PR TITLE
Add shirokov_inverse and hitzer_inverse

### LINKED ISSUE #529: Shirokov inverse?
Hi,

I want to do a symbolic calculation where I have to invert spacetime algebra multivectors that represent transformations, i.e. I use scalar, bivector and pseudoscalar grades only. The goal is to derive a generalization of the transformation [quaternion -> rotation matrix](https://en.wikipedia.org/wiki/Quaternions_and_spatial_rotation#Conversion_to_and_from_the_matrix_representation) for Lorentz transformations.

The `clifford` package can do it using the `.shirokov_inverse()` operation, however using `.inv()` (conventional implementations) does not work. The `galgebra` also struggles upon `u.inv()` because 'TypeError: `u * u.rev()` is not scalar'.

Is there a reason why the Shirokov Inverse is not implemented in `galgebra`? 
The [`clifford` implementation](https://github.com/pygae/clifford/blob/master/clifford/_layout.py#L574) does not look too crazy. I would be happy to give it a shot, if someone is confident that implementing this makes sense also for a symbolic tool like `galgebra`.

Cheers,
Jonas

Minimal working example `galgebra`:
```
import sympy
from galgebra.ga import Ga

ijkl = (i, j, k, l) = sympy.symbols("0 1 2 3", real=True)
algebra = Ga("e_0 e_1 e_2 e_3", g=[1,-1,-1,-1], coords=ijkl)
u = algebra.mv("uS", "scalar") + algebra.mv("uB", "bivector") + algebra.mv("uP", "pseudo")

print(u * u.rev()) # 'scalar + pseudoscalar' 
u.inv() # throws TypeError 'u*u.rev() is not a scalar'
```

Minimal working example `clifford`:
```
import clifford
import numpy as np
LAYOUT, BLADES = clifford.Cl(1, 3)

random = np.random.randn(8)
u = np.zeros(16)
u[0] = random[0]
u[5:11] = random[1:-1]
u[-1] = random[-1]
u = clifford.MultiVector(LAYOUT, value=u)
print(u* ~u) # 'scalar + pseudoscalar'
u.shirokov_inverse() # works
```

### PR DESCRIPTION
**Pull request by spinjo**
Add `shirokov_inverse` and `hitzer_inverse` functions, based on the `clifford` package implementation of the [shirokov_inverse](https://github.com/pygae/clifford/blob/master/clifford/_layout.py#L574) and [hitzer_inverse](https://github.com/pygae/clifford/blob/master/clifford/_layout.py#L592).

closes #529

### CODE REVIEW

**Comment by spinjo** [comment]
I did a few tests on dim=3 and dim=4, but not on dim>=5 because the code was so slow. Happy about ideas on how to speed it up! Plus I think it would make sense to add unit tests for inverse computation.

**Comment by utensil** [comment]
For a first PR, dim 3 to 4 tests suffice. Slowness of GAlgebra above 4 dims are observed in other cases too.

There are a few issues to explore the possibility to speed up GAlgebra, but none is sufficiently mature.

I'll take a look at the code ASAP. Thanks for the PR! ❤️

File: galgebra/mv.py [hunk 1]
```diff
@@ -1359,6 +1359,71 @@ def inv(self) -> 'Mv':
...
+        Iterative algorithm for the inverse following
+        Theorem 4 (page 17) in https://arxiv.org/abs/2005.04015
+        """
+        n = self.Ga.n
+        exponent = (n + 1) // 2
+        N = 2**exponent
+        Uk = self
+        for k in range(1, N):
+            Ck = (N / k) * Uk.scalar()
+            adjU = Ck - Uk
```
Comment (utensil) [inline galgebra/mv.py:1373]: I can't immediately see why you negated it here and negated it back below.

**Comment by utensil** [comment]
Looks correct and good to me, except one comment.

Can you add your tests for dim 3-4 as tests in CI? It's fine to add them as notebooks or vanilla tests.

File: galgebra/mv.py (hunk 1 above)
Comment (spinjo) [reply galgebra/mv.py:1373]: Thats not in line with the clifford code, but in line with the Shirokov paper. The variables adjU and Uk sometimes differ by a sign in the clifford code for some reason.

Comment (utensil) [reply galgebra/mv.py:1373]: 👌

**Comment by EelcoHoogendoorn** [comment]
> I did a few tests on dim=3 and dim=4, but not on dim>=5 because the code was so slow. Happy about ideas on how to speed it up! Plus I think it would make sense to add unit tests for the inverse.

Doing the hitzer ops recursively, rather than greedily forming the product of all involutions, can be much faster. When working recursively the number of components tends to drop dramatically at each step.

**Pull request merged**